alg/gdal_homography.cpp

```diff
@@ -0,0 +1,619 @@
+/*                      GDALComposeHomographies()                      */
+/************************************************************************/
+
+/**
+ * \brief Compose two homographies.
+ *
+ * The resulting homography is the equivalent to padfH1 and then padfH2
+ * being applied to a point.
+ *
+ * @param padfH1 the first homography, nine values.
+ * @param padfH2 the second homography, nine values.
+ * @param padfHOut the output homography, nine values, may safely be the same
+ * array as padfH1 or padfH2.
+ */
+
+void GDALComposeHomographies(const double *padfH1, const double *padfH2,
+                             double *padfHOut)
+
+{
+    double hwrk[9] = {0.0};
+
+    hwrk[1] =
+        padfH2[1] * padfH1[1] + padfH2[2] * padfH1[4] + padfH2[0] * padfH1[7];
+    hwrk[2] =
+        padfH2[1] * padfH1[2] + padfH2[2] * padfH1[5] + padfH2[0] * padfH1[8];
+    hwrk[0] =
+        padfH2[1] * padfH1[0] + padfH2[2] * padfH1[3] + padfH2[0] * padfH1[6];
+
+    hwrk[4] =
+        padfH2[4] * padfH1[1] + padfH2[5] * padfH1[4] + padfH2[3] * padfH1[7];
+    hwrk[5] =
+        padfH2[4] * padfH1[2] + padfH2[5] * padfH1[5] + padfH2[3] * padfH1[8];
+    hwrk[3] =
+        padfH2[4] * padfH1[0] + padfH2[5] * padfH1[3] + padfH2[3] * padfH1[6];
+
+    hwrk[7] =
+        padfH2[7] * padfH1[1] + padfH2[8] * padfH1[4] + padfH2[6] * padfH1[7];
+    hwrk[8] =
+        padfH2[7] * padfH1[2] + padfH2[8] * padfH1[5] + padfH2[6] * padfH1[8];
+    hwrk[6] =
+        padfH2[7] * padfH1[0] + padfH2[8] * padfH1[3] + padfH2[6] * padfH1[6];
+    memcpy(padfHOut, hwrk, sizeof(hwrk));
+}
+
+/************************************************************************/
+/*                       GDALApplyHomography()                        */
+/************************************************************************/
+
+/**
+ * Apply Homography to x/y coordinate.
+ *
+ * Applies the following computation, converting a (pixel, line) coordinate
+ * into a georeferenced (geo_x, geo_y) location.
+ * \code{.c}
+ *  *pdfGeoX = (padfHomography[0] + dfPixel * padfHomography[1]
+ *                                + dfLine  * padfHomography[2]) / 
+ *             (padfHomography[6] + dfPixel * padfHomography[7]
+ *                                + dfLine  * padfHomography[8]);
+ *  *pdfGeoY = (padfHomography[3] + dfPixel * padfHomography[4]
+ *                                + dfLine  * padfHomography[5]) / 
+ *             (padfHomography[6] + dfPixel * padfHomography[7]
+ *                                + dfLine  * padfHomography[8]);
+ * \endcode
+ *
+ * @param padfGeoTransform Nine coefficient Homography to apply.
+ * @param dfPixel Input pixel position.
+ * @param dfLine Input line position.
+ * @param pdfGeoX output location where geo_x (easting/longitude)
+ * location is placed.
+ * @param pdfGeoY output location where geo_y (northing/latitude)
+ * location is placed.
+ */
+
+void CPL_STDCALL GDALApplyHomography(const double *padfHomography,
```

Comment:
CPL_STDCALL is obsolete:
```suggestion
void GDALApplyHomography(const double *padfHomography,
```